core/pkg/server/sender.go

```diff
@@ -1266,7 +1266,13 @@ func (s *Sender) sendRequestLogArtifact(record *service.Record, msg *service.Log
 func (s *Sender) sendRequestDownloadArtifact(record *service.Record, msg *service.DownloadArtifactRequest) {
 	var response service.DownloadArtifactResponse
 
-	if err := artifacts.NewArtifactDownloader(
+	// Perform the nil check on the GraphQL client.
+	if s.graphqlClient == nil {
+		// Offline mode handling:
+		s.logger.Error("sender: sendRequestDownloadArtifact: cannot download artifact in offline mode")
+		response.ErrorMessage = "Artifact downloads are not supported in offline mode. " +
+			"Run 'wandb sync' after the run completes to sync the request."
```

Comment:
why do we need to tell the user about sync here? aren't they trying to download data?

are you just trying to tell them to make sure to sync their data?